src/main/java/core/basesyntax/service/ShopService.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.service;
+
+import core.basesyntax.service.functionalityexpansion.ActivityTypeEnum;
+import core.basesyntax.service.parsefileinfo.FruitInfo;
+import core.basesyntax.service.parsefileinfo.FruitParser;
+import core.basesyntax.service.strategy.ShopActivityStrategy;
+import java.util.Map;
+
+public class ShopService {
+    private final FruitParser fruitParser;
+
+    public ShopService(FruitParser fruitParser) {
+        this.fruitParser = fruitParser;
+    }
+
+    public void doStrategy(Map<ActivityTypeEnum, ShopActivityStrategy> strategyMap,
+                           String[] fileInfoArr) {
+        for (String activityString : fileInfoArr) {
+            FruitInfo fruitInfo = fruitParser.parseActivity(activityString);
```

Comment:
This action should happen in main (see flow diagram). For example:
List<FruitInfo> fruits = fruitParser.parse(rawFruits)
shopService.execute(fruits)